docs/09-英文/02-Producer/05message1.md

```diff
@@ -0,0 +1,173 @@
+      System.out.printf("%s%n", sendResult);
+    }
+    // Close the producer once it is no longer in use
+    producer.shutdown();
+  }
+}
+```
+
+
+
+### 3.2 Asynchronous Sending
+
+![同步发送](../../picture/异步发送.png)
+
+
+Asynchronous sending is a communication method in which the sender sends a message and then sends the next message without waiting for the server to return a response.
+Asynchronous sending requires the implementation of the **Asynchronous Send Callback Interface** (SendCallback).
+:::note
+Asynchronous sending requires the implementation of the **Asynchronous SendCallback Interface**.
+:::
+After sending a message, the sender does not need to wait for a response from the server to send the next message. The sender receives the response from the server through the callback interface and handles the result. Asynchronous sending is generally used in time-consuming and response time sensitive business scenarios. For example, the video upload notifies the start of transcoding service, and notifies the push of transcoding result after transcoding is completed.
+
+The following is the sample code.
+
+``` javascript {16,17}
+public class AsyncProducer {
+  public static void main(String[] args) throws Exception {
+    // Initialize a producer and set the Producer group name
+    DefaultMQProducer producer = new DefaultMQProducer("please_rename_unique_group_name");
+    // Set the address of NameServer
+    producer.setNamesrvAddr("localhost:9876");
+    // Start Producer
+    producer.start();
+    producer.setRetryTimesWhenSendAsyncFailed(0);
+    for (int i = 0; i < 100; i++) {
+      final int index = i;
+      // Create a message and set the topic, tag, body and so on. The tag can be understood as a label to categorize the message, and RocketMQ can filter the tag on the consumer side.
+      Message msg = new Message("TopicTest",
+        "TagA",
+        "Hello world".getBytes(RemotingHelper.DEFAULT_CHARSET));
+      // Send a message asynchronously, the result is returned to the client by callback
+      producer.send(msg, new SendCallback() {
+        @Override
+        public void onSuccess(SendResult sendResult) {
+          System.out.printf("%-10d OK %s %n", index,
+            sendResult.getMsgId());
+        }
+        @Override
+        public void onException(Throwable e) {
+          System.out.printf("%-10d Exception %s %n", index, e);
+          e.printStackTrace();
+        }
+      });
+    }
+    // Close the producer once it is no longer in use
+    producer.shutdown();
+  }
+}
+```
+
+:::note
+The only difference between asynchronous and synchronous sending code is the parameters for calling the sending interface. Asynchronous sending does not wait for the return of the send, instead, the send method needs to be passed into the SendCallback implementation. The SendCallback interface mainly has onSuccess and onException two methods, indicating that the message is sent successfully or failed.
```

Comment:
> The only difference between asynchronous and synchronous sending code is the parameters for calling the sending interface. Asynchronous sending does not wait for the return of the send, instead, the send method needs to be passed into the SendCallback implementation. The SendCallback interface mainly has onSuccess and onException two methods, indicating that the message is sent successfully or failed.

to:

> The only difference between asynchronous and synchronous sending methods is the parameters for calling the sending interface. Asynchronous sending does not wait for the return of send() method, instead, it will carry the SendCallback implementation. The SendCallback interface has two methods (onSuccess and onException), indicating that the message is sent successfully or failed.